.devcontainer/adx/create-adx.sh

```diff
@@ -0,0 +1,131 @@
+
+# Wait Service Principal propagation
+echo "$(tput setaf 2)[4/8] Wait 60 seconds until Service Principal is propagated.$(tput setaf 7)"
+sleep 60
+
+# Assign Service Principal to database permissions
+echo "$(tput setaf 2)[5/8] Assigning database permissions.$(tput setaf 7)"
+az kusto database add-principal \
+    --cluster-name $ADX_CLUSTER_NAME \
+    --database-name $ADX_DB_NAME \
+    --resource-group $RESOURCE_GROUP_NAME \
+    --value name=$SERVICE_PRINCIPAL_DISPLAYNAME fqn=$SERVICE_PRINCIPAL_FQN role="Viewer" type="App"
+
+# Create Storage Account
+echo "$(tput setaf 2)[6/8] Creating Storage Account.$(tput setaf 7)"
+az storage account create \
+    --name $STORAGE_ACCOUNT_NAME \
+    --location $RESOURCE_GROUP_LOCATION \
+    --resource-group $RESOURCE_GROUP_NAME \
+    --sku Standard_LRS
+
+# Connection string used for next az storage commands  
+export AZURE_STORAGE_CONNECTION_STRING=$(az storage account show-connection-string --name $STORAGE_ACCOUNT_NAME --resource-group $RESOURCE_GROUP_NAME --query connectionString -o tsv)
+
+# Create Storage Container
+az storage container create --name $STORAGE_CONTAINER_NAME
+EXPIRY=$(date -u -d '1 hour' '+%Y-%m-%dT%H:%MZ')
+SAS=$(az storage container generate-sas --name $STORAGE_CONTAINER_NAME --permissions lr --expiry $EXPIRY --https-only -o tsv)
+
+# Create Azure Data Explorer Table
+echo "$(tput setaf 2)[7/8] Creating Azure Data Explorer Table.$(tput setaf 7)"
+az storage blob upload --container-name $STORAGE_CONTAINER_NAME --file table.kql
+
+az kusto script create \
+    --cluster-name $ADX_CLUSTER_NAME \
+    --database-name $ADX_DB_NAME \
+    --resource-group $RESOURCE_GROUP_NAME \
+    --script-url "https://$STORAGE_ACCOUNT_NAME.blob.core.windows.net/$STORAGE_CONTAINER_NAME/table.kql" \
+    --script-url-sas-token "?$SAS" \
+    --name "script" \
+    --force-update-tag $(cat /proc/sys/kernel/random/uuid)
+
+# Ingest sample data
+echo "$(tput setaf 2)[8/8] Ingesting Sample Data.$(tput setaf 7)"
+DATA_FILE_NAME=data.json.gz
+curl https://raw.githubusercontent.com/elastic/kibana/v6.8.5/src/server/sample_data/data_sets/flights/flights.json.gz --output $DATA_FILE_NAME
+
+KUSTO_URL=https://$ADX_CLUSTER_NAME.$RESOURCE_GROUP_LOCATION.kusto.windows.net
+TOKEN=$(az account get-access-token --scope "$KUSTO_URL/.default" -o tsv | cut -f1)
+
+curl -X POST "$KUSTO_URL/v1/rest/ingest/$ADX_DB_NAME/kibana_data_flights?streamFormat=json" --header "Authorization: Bearer $TOKEN" --header "Content-Encoding: gzip" --header "Content-Type: multipart/form-data" --data-binary "@$DATA_FILE_NAME"
+rm $DATA_FILE_NAME
+
+# Environment variables needed to run end-to-end tests
+echo "export K2BRIDGE_URL=http://localhost:8080" >> ~/.bashrc 
+echo "export ELASTICSEARCH_URL=http://localhost:9200" >> ~/.bashrc 
+echo "export KUSTO_URI=https://$ADX_CLUSTER_NAME.$RESOURCE_GROUP_LOCATION.kusto.windows.net" >> ~/.bashrc 
+echo "export KUSTO_DB=$ADX_DB_NAME" >> ~/.bashrc 
```

Comment:
We can also set AAD_TOKEN

Reply:
In local development, can be omitted, the script will then use the Azure CLI to generate a token. (I tested and it works fine)
If I set AAD_TOKEN, it will probably need to be renewed if end-to-end tests are launched 2 days later.